DH-10111 Fix missing horizontal scroll bar for single column tables

mattrunyon opened the pull request:
Fixes #446 and related to DH-10111 test failure

This adds a horizontal scroll bar for any single column table which does not fit in the panel fully

Example Python to test with
```python
from deephaven import new_table
from deephaven.column import string_col
import random
import string

letters = string.printable
strings = [''.join(random.choice(letters) for i in range(random.randint(0, 1000))) for j in range(0, 100)]
table1 = new_table([string_col("Random", strings)])
```

Also tested with changing `IrisGridTheme.rowHeaderWidth` to be non-zero

mofojed commented on packages/grid/src/GridMetricCalculator.ts:
```diff
@@ -319,47 +325,54 @@ export class GridMetricCalculator {
-            barHeight - 1
-          )
-        : 0;
+    // How big the horizontal is relative to the horizontal bar
```
We should be doing this for both horizontal and vertical scroll bars. Row height can be resized to be larger than the viewport as well (though not a use case we currently have).

mattrunyon replied:
We are doing this for vertical bar currently, I just didn't rename it like I did horizontal. Do you want me to rename vertical too so it's more clear?

mofojed reviewed (changes requested):
Shift+Wheel (which should scroll horizontally) isn't working. I think need to update Grid.tsx `handleWheel` function

mattrunyon commented:
Updated so vertical scroll uses the same logic as horizontal

Tested by setting a big `rowHeight` in `IrisGridTheme.module.scss`. Also fixed/tested horizontal scroll with `scrollSnapToColumn: true` in `IrisGridTheme.ts`
